backends/ebpf/psa/externs/ebpfPsaCounter.cpp

```diff
@@ -153,6 +187,35 @@ void EBPFCounterPSA::emitMethodInvocation(CodeBuilder* builder, const P4::Extern
     builder->blockEnd(false);
 }
 
+void EBPFCounterPSA::emitDirectMethodInvocation(CodeBuilder* builder,
+                                                const P4::ExternMethod* method,
+                                                cstring valuePtr) {
+    if (method->method->name.name != "count") {
+        ::error(ErrorType::ERR_UNSUPPORTED, "Unexpected method %1%", method->expr);
+        return;
+    }
+    BUG_CHECK(isDirect, "Bad Counter invocation");
```

Comment:
should these be ::error, or are you satisfied with BUG?

Reply:
This situation should never occur, where for DirectCounter emitMethodInvocation is called and for Counter emitDirectMethodInvocation is called. This has more to do with compiler implementation than P4 code, so `BUG` is a better option in my opinion.